Add configuration and recreation instructions (#46)
10up/wp-local-docker-docs: _includes/markdown/Environments.md

A special hostname `all` is available that will start all environments as well as the global services.

## Configure an Environment

An environment's `config/` directory contains settings for various services:

- `elasticsearch/`
    - `elasticsearch.yml`
- `nginx/`
    - `default.conf`
    - `develop.conf`
    - `server.conf`
- `php-fpm/`
    - `docker-php-ext-xdebug.ini` — this is equivalent to `php.ini`, and is where you should put settings that would be found in a file named `php.ini`, such as `max_execution_time`, `memory_limit`, `xdebug.max_nesting_level`, `post_max_size`, `upload_max_filesize`, and similar.
    - `wp-cli.develop.yml`
    - `wp-cli.local.yml`

> After making changes to files in `config/` or to `docker-compose.yml`, you must recreate the containers. Instructions for doing so can be found on this page under [Restart an environment](#restart-an-environment)

Reviewer: ```suggestion
After making changes to files in `config/` or to `docker-compose.yml`, you must restart the containers. Instructions for doing so can be found on this page under [Restart an environment](#restart-an-environment)
```

## Restart an Environment

`10updocker restart <hostname>` will restart all services associated with a preexisting environment.

A special hostname `all` is available that will restart all environments as well as the global services.

> Restarting with `10updocker restart` does not apply configuration changes. If you've made changes to an environment's `docker-compose.yml` or the files in `/config` directory, you will need to stop the environment (but not global services) and then run [`docker-compose up`](https://docs.docker.com/compose/reference/up/) to recreate the containers:

Reviewer: ```suggestion
Restarting with `10updocker restart` does not apply configuration changes. If you've made changes to an environment's `docker-compose.yml` or the files in `/config` directory, you will need to stop the environment (but not global services) and then start it again to make containers re-read changes from the docker-compose file and re-mount configs:
```

Author: A simple `10updocker restart` does not work. The changes are not applied until `docker-compose up` is run.

0. Navigate to the root directory of the install you want to make changes to.
1. Make your changes to the configuration files.
2. Run `10updocker stop all` to stop all environments, including the global services.
3. Run `10updocker start` to get this install and the global services running.

> 4. Run `10updocker stop` to stop this install but not the global services.

Author: The reason I'm having the user go through the dance of `stop all`, `start`, `stop` is because `docker-compose up` fails if the global services are not running. The dance ensures that the global services are running.

It also ensures that no other environments are running. Modifying one environment ought not to modify any others, but I'd like to remove even the possibility of that happening by not having any other environments running. We don't know what else is going on in the user's computer; this helps eliminate any potential sources of interference.

> 5. Run [`docker-compose up`](https://docs.docker.com/compose/reference/up/) to recreate and start this install's docker containers. Wait for it to finish starting; this can be determined by watching the console output. Don't load the site in a browser. If there hasn't been anything new on the console for at least 10 seconds, you can proceed to the next step.
> 6. Kill the newly-started docker process with the `Control-C` keyboard command, or your system's equivalent instruction.
> 7. Run `10updocker restart` to restart the container.

Reviewer: And these steps are not necessary as well:
```suggestion
1. Run `10updocker start` to start containers.
```

Author: These steps _are_ necessary. `10updocker restart` or `10updocker stop; 10updocker start` does not apply the changes. See https://github.com/10up/wp-local-docker-v2/issues/86#issuecomment-852376929

Reviewer 2: @eugene-manuilov is this a bug then? I have also experienced similar behavior with the `restart` command.
